### DISCUSSION TITLE
Commercial Use?

### DISCUSSION THREAD

**Discussion by trpinnovator** [gravitl/netmaker#1566] [Q&A]
First of all thanks for making netmaker.

Since netmaker is under SSPL license, if a commercial company/startup use netmaker for their internal networking, will they be required to release the source code of all of the company products and services using the network? That seems to be the implication of this license.

Some readings:
- https://news.ycombinator.com/item?id=25782471
- https://coralogix.com/blog/elasticsearch-sspl-license-threat-to-business/
- https://opensource.stackexchange.com/questions/8025/difference-between-mongodb-sspl-and-gnu-agpl

As per google `Code released under the [GNU Affero General Public License (AGPL)](http://www.gnu.org/licenses/agpl-3.0.html), the [Open Software License (OSL)](https://opensource.org/licenses/osl-2.1.php), or the Server Side Public License (SSPL) cannot be used in google3 under any circumstances`, see https://opensource.google/documentation/reference/thirdparty/licenses#agpl_affero_gpl_osl_and_sspl_not_allowed.

From my understanding, **ANY** and **ALL work** done while using the netmaker network **must be released under SSPL**. Any thoughts/clarifications on this?

**Comment by trpinnovator** [comment]
@afeiszli @0xdcarns  I found https://docs.netmaker.org/support.html#why-the-sspl-license and now understand the intention was to ensure the long-term viability of the project -- which I strongly support.

Given that, may I politely ask if you have considered using instead [Mozilla Public License v2.0](https://www.mozilla.org/en-US/MPL/2.0/FAQ/)?

Hashicorp uses MPL v2.0 in its core products: nomad, vault, consul, packer,.. https://github.com/hashicorp. They were able to make an enterprise offering and so far it seems no one has attempted to make a SAAS offering of their products.

This is just a thought/suggestion. Regardless of your decision, I hope Netmaker as a product/company succeeds and make it big. Thanks.

**Reply by afeiszli** [reply]
Thanks for this input. We review (and debate) or license choices internally every so often. We are still considering if a license change makes sense for us right now, and will definitely take this into consideration.
